### PR TITLE
Protect: Add IP block and allow list toggles

### PR DESCRIPTION
**Pull request by dkmyta**
### Description
Updates the Protect Firewall UI to allow the IP block and allow lists to be toggled independently.

## Proposed changes:

* Makes use of the new `jetpack_waf_ip_allow_list_enabled` / `jetpack_waf_ip_block_list_enabled` settings, replacing `jetpack_waf_ip_list`
* Removes the dedicated page manual rules and adds independent toggles for each IP list to the primary Firewall page
* Modifies sections dependent on the original setting

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1262 

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch
* Start up JT, and enable Protect
* Proceed to Firewall settings and toggle the various options in different combinations
* Ensure the settings all perform as expected and that no regressions are introduced
* Manually set `Waf_Runner::is_unsupported_environment` to false and verify that the BFP allow list still functions
* Upgrade and downgrade and ensure all expected settings are still accessible and functional
* Verify that a blocklisted IP cannot make requests or access the site admin when BFP is enabled
* Ensure that the blocked IP can once again do so once whitelisted
* Ensure cross compatibility exists when Jetpack is enabled and the settings are toggled from that end
* Ensure backward compatibility exists when the update is applied to an existing site with the prior settings in place

## Screenshots

Enabled:
Screen Shot 2024-07-17 at 08 55 18

Disabled:
Screen Shot 2024-07-17 at 08 57 01

WAF Disabled:
Screen Shot 2024-07-17 at 08 58 07

Unsupported Env:
Screen Shot 2024-07-17 at 09 04 33

### CODE REVIEW

File: projects/packages/waf/src/class-rest-controller.php [hunk 1]
```diff
@@ -170,8 +170,8 @@ public static function update_waf( $request ) {
 			}
 		}
 
-		// Only attempt to update the WAF if the module is supported
-		if ( Waf_Runner::is_supported_environment() ) {
+		// Only attempt to update the WAF if the module is supported and enabled.
+		if ( Waf_Runner::is_supported_environment() && Waf_Runner::is_enabled() ) {
```
Comment (nateweller) [inline projects/packages/waf/src/class-rest-controller.php]: I'm wondering if https://github.com/Automattic/jetpack/pull/38326 makes this change unnecessary, I will double check.

**Comment by nateweller** [comment]
Suggestion – 

Now that the waf package supports separate toggling of the allow/block list, there's opportunity here to simplify our code and remove the duplication of the allow list.

We could get rid of the `manualRulesSettings` and "manual rules" concept entirely, and output each toggle individually.

i.e.
```
<div className={ styles[ 'toggle-wrapper' ] }>
    { wafSupported && automaticRulesSettings }
    { bruteForceProtectionSettings }
    { wafSupported && ipBlockListSettings }
    { ipAllowListSettings }
</div>
```

**Comment by nateweller** [comment]
What do you think about this positioning for the "add IP to allow list" button?

Screenshot 2024-07-15 at 1 30 45 PM

File: projects/packages/waf/src/class-rest-controller.php (hunk 1 above)
Comment (dkmyta) [reply projects/packages/waf/src/class-rest-controller.php]: Yep, just merged the change and it appears to no longer be an issue! I'll revert this now.

File: projects/plugins/protect/src/js/routes/firewall/index.jsx [hunk 2]
```diff
@@ -831,7 +794,13 @@ const FirewallPage = () => {
 			<ConnectedFirewallHeader />
 			<Container className={ styles.container } horizontalSpacing={ 8 } horizontalGap={ 4 }>
 				{ wafSupported && ! isEnabled && <Col>{ moduleDisabledNotice } </Col> }
-				<Col>{ ! showManualRules ? mainSettings : manualRulesSettings }</Col>
+				<Col>
+					<div className={ styles[ 'toggle-wrapper' ] }>
+						{ wafSupported && automaticRulesSettings }
+						{ bruteForceProtectionSettings }
```
Comment (nateweller) [inline projects/plugins/protect/src/js/routes/firewall/index.jsx:798]: Minor – can we move the `ipAllowList` out of `bruteForceProtectionSettings` and have it rendered here? Will improve readability as the allow list is contextually applicable at this higher level.

i.e.
```
{ wafSupported && automaticRulesSettings }
{ bruteForceProtectionSettings }
{ ipAllowListSettings }
{ wafSupported && ipBlockListSettings }
```

File: projects/plugins/protect/src/js/routes/firewall/index.jsx [hunk 3]
```diff
@@ -576,54 +643,75 @@ const FirewallPage = () => {
...
 		<>
-			<div className={ styles[ 'brute-force-rules-section' ] }>
-				<Textarea
-					id="jetpack_waf_ip_allow_list"
-					label={ __( 'Always allowed IP addresses', 'jetpack-protect' ) }
-					description={
-						<>
-							<Text mb={ 1 }>
-								{ __(
-									"IP addresses added to this list will never be blocked by Jetpack's brute force protection.",
```
Comment (nateweller) [inline projects/plugins/protect/src/js/routes/firewall/index.jsx:589]: Suggestion – can we keep the description for the allow list, i.e. "IP addresses added to this list will never be blocked by Jetpack's security features."

**Comment by nateweller** [comment]
Code looks great, tests great as well!

What do you think about swapping the order of the allow/block list? My thinking being, the block list is more related to the firewall features, and the allow list is a final exception to any of the features on the page, so it might be suited at the bottom.

**Review by nateweller — DISMISSED** [review]
Looks good and tests well! LGTM 👍 👍

**Review by nateweller — APPROVED** [review]
LGTM! 🚢

**Pull request merged**